client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -92,7 +92,9 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 						<div className={ classnames( flow.name, flow.classnames, kebabCase( path ) ) }>
 							<ProgressBar
 								// eslint-disable-next-line wpcalypso/jsx-classname-namespace
-								className="flow-progress"
+								className={ classnames( 'flow-progress', {
+									'is-fixed': stepData?.isGeneratedDesignsView && path === 'designSetup',
```

Comment:
I think using `stepData` is not necessary and we still need to create another PR to remove this condition after we ship the UDP.

How about just setting the class inside `useEffect` directly and avoid `FlowRenderer` re-rendering (but this is not important as UDP will ship in the next few days)? For example,

```js
// site-setup-design-picker.tsx
useEffect( () => {
  const flowProgressElement = document.querySelector( '.flow-progress' );
  const targetClassName = 'is-fixed';
  if ( showGeneratedDesigns ) {
    flowProgressElement.classList.add( targetClassName );
  } else {
    flowProgressElement.classList.remove( targetClassName );
  }

  // Not sure whether to remove the class in the clean-up function.
  // Or we can just do it in the `handleBackClick` function
  return () => {
    flowProgressElement.classList.remove( targetClassName );
  };
}, [ showGeneratedDesigns ] );
```

But we might still need to remove `.is-fixed` style from `client/landing/stepper/declarative-flow/internals/global.scss` after UDP is shipped

What do you think?

Reply:
If querySelector is accepted by the team, then yes, I'll go with this suggestion. But as you pointed out this issue has become an unnecessary effort already :sweat_smile:, so I think either we leave this away or else revert the last PR https://github.com/Automattic/wp-calypso/pull/66056

cc: @taipeicoder

Reply:
> so I think either we leave this away or else revert the last PR #66056

Yes, that makes sense to me. I'm okay with the current way and then we can just revert this one to clean up when UDP is stable

Reply:
@bangank36 could you create an issue to remove `is-fixed` so that we don't forget about this?

Reply:
Sure https://github.com/Automattic/wp-calypso/issues/66432